### PR TITLE
CHAT-136 Add NCA functionality, and infer route from data

### PR DESCRIPTION
**Pull request by roninsightrx**
- This adds functionality for parsing SDTM to data suitable for IQnca package.
- It also infers the route from the data, both for NCA and modeling analysis, instead of having user supply the route.
- Also abstracted the data reading functions a bit more.

### CODE REVIEW

File: R/parse_data_for_modelfit.R [hunk 1]
```diff
@@ -318,11 +329,7 @@ parse_data_for_modelfit <- function(data) {
         SEX == "M" ~ 1,
         SEX == "F" ~ 2,
         TRUE ~ 3
-      ),
-      ROUTE = "oral",    # TODO: hardcoded for now
```
Comment (roninsightrx) [inline R/parse_data_for_modelfit.R:322]: we actually didn't need it here, this is the temporary table holding the covariates.

File: R/parse_data_for_modelfit.R [hunk 2]
```diff
@@ -221,6 +228,10 @@ parse_data_for_modelfit <- function(data) {
         is.na(NFRLT_prev) ~ NFRLT - min_NFRLT,
         TRUE ~ NFRLT - NFRLT_prev
       )
+    ) %>%
+    dplyr::mutate(
+      ROUTE = EXROUTE, 
+      FORM = EXDOSFRM
```
Comment (roninsightrx) [inline R/parse_data_for_modelfit.R:234]: we want to add ROUTE and FORM to all dosing rows. For all observations the values in these columsn will be NA, which is intended.

File: R/parse_data_for_nca.R [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    ADUR = ifelse(route == "iv", NA, 0), # TODO: handle for IV drugs
+    NDUR = ifelse(route == "iv", NA, 0)  # TODO: handle for IV drugs
+  )
+  
+  ## Some cohorts we don't want to analyze in NCA, e.g. because they're not
+  ## actually in the study or did not receive the drug.
+  remove_cohorts <- c("Screen Failure", "Placebo")
+
+  merged_data <- data$pc %>%
+    setNames(toupper(names(.))) %>%
```
Comment (karawoo) [inline R/parse_data_for_nca.R:31]:
```suggestion
    stats::setNames(toupper(names(.))) %>%
```

File: R/parse_data_for_nca.R [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    merge(
+      data$dm %>% # join with study-arm data
+      setNames(toupper(names(.))) %>%
```
Comment (karawoo) [inline R/parse_data_for_nca.R:34]:
```suggestion
      stats::setNames(toupper(names(.))) %>%
```

**Review by karawoo — CHANGES_REQUESTED** [review]
A couple of small comments, there are also some documentation mismatches right now but that's not an issue that would affect running the code.

**Review by karawoo — APPROVED** [review]

**Pull request merged**